test/support/spec.rb

```diff
@@ -1434,3 +1436,60 @@ def run_specs
 end
 
 at_exit { run_specs }
+
+
+# NATFIXME method to wrap spec-matchers that are known to fail and mark them
+# as "fix this later".
+#
+# Can be used in the following ways:
+#
+# 1) hide any StandardError inside of the block
+# NATFIXME "some description" do
+#   some_spec.should be "bad"
+# end
+#
+# 2) Hide any error matching a klass
+# NATFIXME "some description", klass: NoMethodError do
+#   some_spec.foo.should be "also bad"
+# end
+#
+# 3) Hide any error matching a klass and with error-message match
+# NATFIXME "some description", klass: NoMethodError, match: /method foo undefined/ do
+#   some_spec.foo.should be "also bad"
+# end
+#
+# If the code inside the block does not cause the error or the error it caues does not match the klass (or klass/match) then
+# NatalieFixMeException will be raised.
+#
+def NATFIXME(description, klass: nil, match: nil)
+  raise SpecFailedException, "NATFIXME requires a block" unless block_given?
+  klass ||= StandardError
+  match = case match
+          when String then Regexp.new(match)
```

Comment:
This does not escape the regex characters, so that it probably not what you want:
```
nat> match = Regexp.new('foo.')
/foo./
nat> match.match?('foox')
true
```
It is possible to use quoting:
```
nat> match = Regexp.new(Regexp.quote('foo.'))
/foo\./
nat> match.match?('foox')
false
nat> match.match?('foo.')
```

But I would personally just rewrite it to some lambdas:
```ruby
case match
when String then ->(message) { message.include?(match) }
when Regexp then ->(message) { match.match?(message) }
when nil then ->(_) { true }
else raise ArgumentError, "match must be nil, Regexp or String"
end
```
And of course update the caller too (which can't be nil at that line, so we can drop the first check too):
```ruby
if match.call(ex.message)
```

Reply:
Awesome idea, I updated and pushed changes based on your example above.